boards/arm/stm32f412g_disco/stm32f412g_disco_defconfig

```diff
@@ -0,0 +1,45 @@
+CONFIG_ARM=y
+CONFIG_BOARD_STM32F412G_DISCO=y
+CONFIG_SOC_SERIES_STM32F4X=y
+CONFIG_SOC_STM32F412ZG=y
+# 96MHz system clock (highest value to get a precise USB clock)
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=96000000
+CONFIG_SYS_CLOCK_TICKS_PER_SEC=1000
+
+# Enable MPU
+CONFIG_STM32_ARM_MPU_ENABLE=y
+
+# enable USART2 - passthrough to STLINK v2 connector
+CONFIG_UART_STM32_PORT_2=y
+
+# enable pinmux
+CONFIG_PINMUX=y
+
+# enable GPIOs
+CONFIG_GPIO=y
+CONFIG_GPIO_STM32=y
+CONFIG_GPIO_STM32_PORTA=y
+CONFIG_GPIO_STM32_PORTB=y
+CONFIG_GPIO_STM32_PORTC=y
+CONFIG_GPIO_STM32_PORTD=y
+CONFIG_GPIO_STM32_PORTE=y
+
+# clock configuration
+CONFIG_CLOCK_CONTROL=y
+
+# Clock configuration for Cube Clock control driver
+CONFIG_CLOCK_STM32_HSE_CLOCK=8000000
+CONFIG_CLOCK_STM32_SYSCLK_SRC_PLL=y
+# use HSE as PLL input
+CONFIG_CLOCK_STM32_PLL_SRC_HSE=y
+# however, the board does not have an external oscillator, so just use
+# the 8MHz clock signal coming from integrated STLink
+CONFIG_CLOCK_STM32_HSE_BYPASS=y
+# produce 96MHz clock at PLL output
```

Comment:
The maximum frequency of the STM32F412GZ is 100 MHz. Please also check the PPL_x values and the prescalers below. Check the RCC section of the SoC's reference manual to calculate the correct values.